### PR TITLE
chore: failing test and deprecated code used

### PR DESCRIPTION
**Pull request by villaleo**
- The test `formatCompactCurrency` has an edge case where an amount like 1,000 will be formatted as `$1.0K` instead of `$1K`. I didn't see a better way to go about fixing this, so I used a regex to `replace` the first instance of `.0`.
- `AssetWithRelation` is deprecated, but still used in `@/components/assets/AssetForm.tsx`.

### CODE REVIEW

File: src/lib/utils/formatters.ts [hunk 1]
```diff
@@ -54,7 +54,9 @@ export function formatCompactCurrency(value: number): string {
     currency: 'USD',
     notation: 'compact',
     maximumFractionDigits: 1,
-  }).format(value)
+  })
+    .format(value)
+    .replace(/.0/g, '')
```
Comment (elimanzo) [inline src/lib/utils/formatters.ts]: Can you 2x check that this is needed? [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#maximumfractiondigits) should handle this since it's rounding off the first digit after the `.`  and `1000` should be `1k` 

we do have `formatCompactNumber` that just formats decimals that doesn't use these formatting options, but that is intended. 

For the regex: the `.` is unescaped, matches any char, and using `/g` can be risky. I believe a value like $10.0M would become $1M. something like `/\.0$/` can remove the trailing zero (PS it's been a while since i've done regex haha, but 2x check if the 1st solution is still okay above).

**Review by elimanzo — APPROVED** [review]
This is good thanks! I have a 1 comment and it should be ready to go!

File: src/lib/utils/formatters.ts (hunk 1 above)
Comment (villaleo) [reply src/lib/utils/formatters.ts]: I got it, thanks Eli! We just needed to add `minimumFractionDigits: 0` since we've set `maximumFractionDigits: 1`. No need for regex (although, I tested on [regex101](https://regex101.com/substitution) and `/\.0/` would suffice in this case).
